lib/data_migrator/offence_unique_code_migrator.rb

```diff
@@ -47,5 +56,20 @@ def generator(offence = nil)
         @generator
       end
     end
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    def output(code, offence, format)
```

Comment:
**[not a blocker]** 💭 Consider the following approach to remove the use of extensive case statements (which are considered a bad practice, hence the `CyclomaticComplexity` alert):

```ruby
def output(code, offence, format)
  message = {
    sql: "UPDATE offences SET unique_code = #{code} WHERE id = #{offence[:id]}".yellow,
    diff: offence[:unique_code].eql?(code) ? nil : offence[:description].concat("\n -#{offence[:unique_code]}".red).concat("\n +#{code} ".green)
  }[format.downcase.to_sym]
    text: "-- [would have] updated #{offence[:description]},#{offence[:band] || offence[:class_letter]}".white.concat(" unique_code: #{offence[:unique_code]} --> #{code}".green)
  puts message unless message.blank?
end
```

Reply:
that looks a lot more complex and more work for the method as it has to process more. A simple case statement may have a few more paths through it re `Metrics/CyclomaticComplexity` but is more humanly readable. for a laugh read http://tmont.com/blargh/2010/11/cyclomatic-complexity-is-worthless